editor/hooks/use-pending-callback.tsx

```
@@ -0,0 +1,72 @@
+ */
+interface PendingCallbackState<T extends Function = Function> {
+  /** The callback function to be executed */
+  callback: T | null;
+  /** Whether the callback is currently being executed */
+  isExecuting: boolean;
+}
+
+/**
+ * Handlers for managing a pending callback
+ */
+interface PendingCallbackHandlers<T extends Function = Function> {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)